src/main/java/org/opentripplanner/middleware/utils/HttpUtils.java

```diff
@@ -57,40 +66,54 @@ public static String httpRequest(URI uri, int connectionTimeout, REQUEST_METHOD
             httpRequestBuilder.GET();
         }
 
+        if (method.equals(REQUEST_METHOD.DELETE)) {
+            httpRequestBuilder.DELETE();
+        }
+
         if (method.equals(REQUEST_METHOD.POST)) {
             httpRequestBuilder.POST(HttpRequest
                 .BodyPublishers
                 .ofString(bodyContent));
         }
 
         if (headers != null && !headers.isEmpty()) {
-            for(Map.Entry<String, String> e: headers.entrySet()){
+            for (Map.Entry<String, String> e : headers.entrySet()) {
                 httpRequestBuilder.setHeader(e.getKey(), e.getValue());
             }
         }
 
         HttpRequest request = httpRequestBuilder.build();
 
         try {
-            // raw response
-            HttpResponse<String> httpResponse = client.send(request, java.net.http.HttpResponse.BodyHandlers.ofString());
-            return httpResponse.body();
+            return client.send(request, java.net.http.HttpResponse.BodyHandlers.ofString());
         } catch (InterruptedException | IOException e) {
             BugsnagReporter.reportErrorToBugsnag("Error requesting data from URI", uri, e);
         }
 
         return null;
     }
 
-
     /**
      * Get entity attribute value from request. If nulls are not allowed, halt with error message.
      */
-    public static String getRequiredParamFromRequest(Request req, String paramName, boolean allowNull) {
+    public static String getRequiredQueryParamFromRequest(Request req, String paramName, boolean allowNull) {
         String paramValue = req.queryParams(paramName);
         if (paramValue == null && !allowNull) {
             logMessageAndHalt(req, HttpStatus.BAD_REQUEST_400, "The parameter name " + paramName + " must be provided.");
         }
         return paramValue;
     }
+
+    /**
+     * Get a request parameter value.
+     * This method will halt the request if paramName is not provided in the request.
+     */
+    public static String getRequiredParamFromRequest(Request req, String paramName) {
+        String paramValue = req.params(paramName);
+        if (paramValue == null) {
+            logMessageAndHalt(req, HttpStatus.BAD_REQUEST_400, "Must provide parameter name.");
```

Comment:
Should this return the missing paramName?

Reply:
Updated.